pkg/controller/nodelifecycle/scheduler/taint_manager_test.go

```diff
@@ -314,22 +382,24 @@ func TestUpdatePod(t *testing.T) {
 
 		holder.setPod(item.prevPod)
 		controller.PodUpdated(nil, item.prevPod)
+
 		fakeClientset.ClearActions()
-		time.Sleep(timeForControllerToProgress)
+
 		holder.setPod(item.newPod)
 		controller.PodUpdated(item.prevPod, item.newPod)
-		// wait a bit
-		time.Sleep(timeForControllerToProgress)
-		if item.additionalSleep > 0 {
-			time.Sleep(item.additionalSleep)
+
+		actions := fakeClientset.Actions()
+		if item.expectDelete {
```

Comment:
Instead of the if, invoke the wait function with the information for which pods the delete is expected.